src/internal/task/queue.go

```diff
@@ -10,7 +12,9 @@ type Queue struct {
 
 // Push a task onto the queue.
 func (q *Queue) Push(t *Task) {
+	i := interrupt.Disable()
```

Comment:
It might be more efficient to put all these locks where the queue/stack are used.